### ISSUE TITLE
[0.9.4.1] fill only works if there is at least one data point

### ISSUE THREAD

**Issue by beckettsean** [influxdata/influxdb#4273]
I'm wondering why the `fill` function doesn't do anything if all buckets are `nil`. Instead of getting the buckets returned with whatever `fill` specifies there's just an empty response.

For example, here's the data:

```
> insert fillfunc,a=1 value=1 1443547642000000000
> insert fillfunc,a=1 value=2 1443547643000000000
> insert fillfunc,a=1 value=3 1443547644000000000
> insert fillfunc,a=1 value=4 1443547645000000000
```

And here's the successful query with `fill`:

```
> select count(value) from fillfunc where time > 1443547641000000000 and time < 1443547652000000000 and a='1' group by time(1s) fill(0)
name: fillfunc
--------------
time			count
2015-09-29T17:27:21Z	0
2015-09-29T17:27:22Z	1
2015-09-29T17:27:23Z	1
2015-09-29T17:27:24Z	1
2015-09-29T17:27:25Z	1
2015-09-29T17:27:26Z	0
2015-09-29T17:27:27Z	0
2015-09-29T17:27:28Z	0
2015-09-29T17:27:29Z	0
2015-09-29T17:27:30Z	0
2015-09-29T17:27:31Z	0
```

but if there's no matching point at all, the query just returns an empty response:

```
> select count(value) from fillfunc where time > 1443547641000000000 and time < 1443547652000000000 and a='2' group by time(1s) fill(0)
> 
```

Here's the `curl` output showing the empty response:

```
$ curl -G 'http://localhost:8086/query?db=mydb&rp=default' --data-urlencode "q=select count(value) from fillfunc where time > 1443547641000000000 and time < 1443547652000000000 and a='2' group by time(1s) fill(0)"
{"results":[{}]}
```

**Comment by beckettsean** [comment]
inspired by https://github.com/influxdb/influxdb/issues/4247#issuecomment-143895025, which has the actual use case for `fill(0)` returning values per bucket in this case.

**Comment by cheppsn** [comment]
We also noticed this behavior in 0.10.3 and a worsening of the situation in the more recent 0.12. 

In 0.10.3 we were able to provide fill(0) in which case the expected number of buckets would be returned, all initialized with value 0. Using fill(null) or fill(previous) didn't work either, as described above. However, in 0.12.0 we now get a completely empty result, even if fill(0) is provided.

When querying data with group by(time) and fill(x) inside a completely data-free interval, we'd expect Influx to return values as specified by fill(x) for every calculated bucket, unless fill(none) is used.

It'd be great if this could be fixed with 0.13, currently we have to construct the null-valued buckets in our application logic.

**Comment by jsternberg** [comment]
Unfortunately, I'm going to need to close this. Fill is unable to figure out what buckets to fill due to infinite number of possible bucket combinations when tags are involved. If fill only had to fill in a single series with a single measurement (no tags), this would be possible. You can have this type of query:

```
SELECT mean(value) FROM cpu WHERE time > now() - 10m GROUP BY time(1m), host
```

We can't possibly find all of the possible values that `host` can have with fill. I do think we're going to need to think of something at some point about this, but I'm going to close this for now. This is similar to #6412 and I'll keep that one open instead.

Thanks.
